common/metrics/interfaces.go

```diff
@@ -43,5 +43,8 @@ type (
 		UpdateGauge(scope int, gauge int, value float64)
 		// Tagged returns a client that adds the given tags to all metrics
 		Tagged(tags map[string]string) Client
+		// Scope return an internal scope that can be used to add additional
+		// information to metrics
+		Scope(scope int) Scope
```

Comment:
(1) can you get rid of the free style Tagged(tags map[string]string) method in favor of the strongly typed tags ? I don't expect a lot of places we use this today
(2) How about making the signature of Scope as  Scope(scope int, tags ..Tag) ? Any caller that calls Scope() almost always will need a sub-scope with some tags. So, taking the tags as input here saves additional api call / data copy later

Reply:
Got rid of tagged. And change the Scope method to take tags. The actual Tagged support is still in the underlying scope returned though.

Reply:
👍